clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystemUtils.java

```diff
@@ -180,7 +169,7 @@ public static long persistFile(TachyonFileSystem tfs, TachyonFile file, FileInfo
       closer.close();
     }
     // Tell the master to mark the file as persisted
-    tfs.setState(file, (new SetStateOptions.Builder()).setPersisted(true).build());
+    fs.setAttribute(uri, SetAttributeOptions.defaults().setPersisted(true));
```

Comment:
The defaults for `SetAttributeOptions` are unset attributes, so it will only update setting persisted to true and retain the previous values for others.